server/src/test/java/org/spine3/server/event/EventStoreShould.java

```diff
@@ -105,9 +108,78 @@ public void read_events_by_time_bounds() {
         assertEquals(eventInPresent, event);
     }
 
-    private static Event projectCreated(Timestamp timestamp) {
+    @Test
+    public void read_events_by_time_and_type() {
+        final Duration delta = Durations2.seconds(111);
+        final Timestamp present = getCurrentTime();
+        final Timestamp past = subtract(present, delta);
+        final Timestamp future = add(present, delta);
+
+        final Event eventInPast = taskAdded(past);
+        final Event eventInPresent = projectCreated(present);
+        final Event eventInFuture = taskAdded(future);
+
+        eventStore.append(eventInPast);
+        eventStore.append(eventInPresent);
+        eventStore.append(eventInFuture);
+
+        final EventFilter taskAddedType = EventFilter.newBuilder()
+                                                       .setEventType(of(TaskAdded.class).value())
+                                                       .build();
+        final EventStreamQuery query = EventStreamQuery.newBuilder()
+                                                       .setAfter(past)
+                                                       .addFilter(taskAddedType)
+                                                       .build();
+        final AtomicBoolean done = new AtomicBoolean(false);
+        final Collection<Event> resultEvents = Collections.synchronizedSet(new HashSet<Event>());
```

Comment:
Please try [this](https://google.github.io/guava/releases/21.0/api/docs/com/google/common/collect/Sets.html#newConcurrentHashSet--) instead. 

Same below.